config/routes.rb

```diff
@@ -2,12 +2,16 @@
   root 'homes#index'
   devise_for :users
 
-  resources :vendors
+  resources :vendors, only: [:index, :show, :edit, :update]
 
   resources :vendors do
     resources :locations
   end
 
+  resources :vendors, only: [:index, :show] do
+    resources :reviews
+  end
+
   resources :locations
```

Comment:
Okay, so you guys have a bunch of duplicate resources listed here. You only need to write `resources :vendors, only: [...]` once, and then put all the nested items inside it. If you have a single line that just says `resources vendors` with no `only`, it'll do the work of creating all the routes and then all of the other `only`s you wrote will be to no effect. 

Also, is it intentional to have `locations` listed twice, once nested and once not nested? That'll mean that ever single `location` action will be accessible both as a nested and non-nested route - seems a little excessive. Come to think of it, are you using location at all right now? If not, just get rid of it! (I know that's probably something that I just didn't catch from earlier, but still worth fixing :)